plugins/auto-sizes/includes/improve-calculate-sizes.php

```diff
@@ -288,15 +296,43 @@ function auto_sizes_filter_render_block_context( array $context, array $block ):
 	);
 
 	if ( in_array( $block['blockName'], $provider_blocks, true ) ) {
-		$alignment = $block['attrs']['align'] ?? '';
+		// Normalize default alignment values.
+		$alignment = isset( $block['attrs']['align'] ) && '' !== $block['attrs']['align'] ? $block['attrs']['align'] : 'default';
+		// Use the defined constant for constraints.
+		$constraints = AUTO_SIZES_CONSTRAINTS;
 
-		// If the container block doesn't have alignment, it's assumed to be 'default'.
-		if ( '' === $alignment ) {
-			$context['max_alignment'] = 'default';
-		} elseif ( 'wide' === $alignment ) {
-			$context['max_alignment'] = 'wide';
+		if ( 'default' === $alignment || 'wide' === $alignment ) {
+			$context['max_alignment'] = $constraints[ $context['max_alignment'] ] > $constraints[ $alignment ] ? $context['max_alignment'] : $alignment;
 		}
 	}
 
+	if ( 'core/columns' === $block['blockName'] ) {
+		// This is a special context key just to pass to the child 'core/column' block.
+		$context['column_count'] = count( $block['innerBlocks'] );
+	}
+
+	if ( 'core/column' === $block['blockName'] ) {
+		$found_image_block = wp_get_first_block( $block['innerBlocks'], 'core/image' );
+		$found_cover_block = wp_get_first_block( $block['innerBlocks'], 'core/cover' );
+		if ( count( $found_image_block ) > 0 || count( $found_cover_block ) > 0 ) {
+			// Get column width, if explicitly set.
+			if ( isset( $block['attrs']['width'] ) && '' !== $block['attrs']['width'] ) {
+				$current_width = floatval( rtrim( $block['attrs']['width'], '%' ) ) / 100;
+			} elseif ( isset( $parent_block->context['column_count'] ) && $parent_block->context['column_count'] ) {
+				// Default to equally divided width if not explicitly set.
+				$current_width = 1 / $parent_block->context['column_count'];
+			} else {
+				// Full width fallback.
+				$current_width = 1;
+			}
+
+			// Multiply with parent's width if available.
+			if ( isset( $parent_block->context['column_width'] ) ) {
+				$context['column_width'] = $parent_block->context['column_width'] * $current_width;
+			} else {
+				$context['column_width'] = $current_width;
+			}
```

Comment:
Do we want to stick with the `column_width` name for this context key? It's currently tied only to `core/columns` and `core/column` blocks, but technically speaking it more broadly indicates the current container's relative width compared to whatever the outer content width is based on the alignment (e.g. content, wide, or full).

Maybe we should use a more generic term to clarify that this is not only relevant for the `core/column` block, e.g. `container_width`, or more verbosely `container_relative_width`?

Reply:
Updated in https://github.com/WordPress/performance/pull/1795/commits/75d689b6a417bc8f2eba07e9383ff75511b201fd